### DISCUSSION TITLE
general string format?

### DISCUSSION THREAD

**Discussion by retsyo** [arturo-lang/arturo#1270] [General]
we can use 
```rebol
pi: 3.1415

print ["pi = " to :string .format: ".3f" pi]
```
to print
```
pi =  3.142
```

but we can't use
```rebol
print [to :string .format: "pi = .3f" pi]
```
which is, however, nice for complex string. So my question is, how to do such C's `printf` in arturo?

Thanks

**Comment by RickBarretto** [comment]
You're right, now the equivalent is the `to` or `as`.
For floating-point formatting, you can use `to :string .format: ".2f" ...` as for `:date`s as well.
For binary, hex, decimal and others, you can use `as` for it.

## Examples

```red
print -> to :string .format: "MMM dd yy" now        ; => Sep 03 23
print -> "$" ++ to :string .format: ".2f" 15.05648  ; => $15.05
print -> "x" ++ as.hex 15     ; => xf
print -> "b" ++ as.binary 14  ; => b1110
```

## Ideas

I do agree with you and new ways to formatting would be good, something inside the string interpolation as Python does now, or some C-ish formatting.

I think a `print.format: "{.2f} {MMM dd yy} {as.hex}" [values ...]` is great too. For now, the project has other priorities, but this idea would be good in a near future.
@drkameleon, what do you think?
